Mathlib/CategoryTheory/Limits/Shapes/RegularMono.lean

```diff
@@ -57,6 +59,47 @@ attribute [reassoc] RegularMono.w
 instance (priority := 100) RegularMono.mono (f : X ⟶ Y) [RegularMono f] : Mono f :=
   mono_of_isLimit_fork RegularMono.isLimit
 
+/-- Every isomorphism is a regular monomorphism. -/
+instance (priority := 100) RegularMono.ofIso (f : X ⟶ Y) [IsIso f] : RegularMono f where
+  Z := Y
+  left := 𝟙 Y
+  right := 𝟙 Y
+  isLimit := Fork.IsLimit.mk _ (fun s ↦ s.ι ≫ inv f) (by simp) fun s m w ↦ by
+    rw [IsIso.eq_comp_inv, ← w]; simp
+
+/-- Regular monomorphisms are preserved when precomposing with an isomorphism. -/
+instance (priority := 100) RegularMono.ofIsoCompRegularMono {Z}
+    (f : Y ⟶ Z) [h : RegularMono f] (g : X ⟶ Y) [IsIso g] : RegularMono (g ≫ f) where
```

Comment:
Could you prove more generally that if `Arrow.mk f` is isomorphic to `Arrow.mk g`, then `RegularMono f` implies `RegularMono g`? (This way, we could deduce that `MorphismProperty` respects isomorphisms.)